src/argilla/client/feedback/metrics/agreement_metrics.py

```diff
@@ -0,0 +1,273 @@
+#  limitations under the License.
+
+"""This module contains metrics to gather information related to inter-Annotator agreement. """
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Any, Callable, Hashable, List, Tuple, Union
+
+from nltk.metrics.agreement import AnnotationTask as NLTKAnnotationTask
+from nltk.metrics.distance import binary_distance, edit_distance, interval_distance
+from pydantic import BaseModel
+
+from argilla.client.feedback.schemas import (
+    LabelQuestion,
+    MultiLabelQuestion,
+    RankingQuestion,
+    RatingQuestion,
+    TextQuestion,
+)
+from argilla.client.feedback.schemas.enums import ResponseStatusFilter
+
+if TYPE_CHECKING:
+    from argilla.client.feedback.dataset import FeedbackDataset
+    from argilla.client.feedback.dataset.remote.dataset import RemoteFeedbackDataset
+
+
+# Expected format for the nltk's AnnotationTask
+FormattedResponses = List[Tuple[Any, Hashable, Hashable]]
+
+
+class AgreementMetricResult(BaseModel):
+    """Container for the result of an agreement metric.
+
+    It contains two fields, `metric_name` and `result` with the value of the metric.
+    """
+
+    metric_name: str
+    result: float
+
+
+def prepare_dataset_for_annotation_task(
+    dataset: Union["FeedbackDataset", "RemoteFeedbackDataset"], question_name: str
+) -> FormattedResponses:
+    """Helper function to prepare the dataset for the nltk's AnnotationTask.
+
+    The AnnotationTask class from nltk expects the data to be formatted as a list
+    of tuples, each containing the annotator id, the task id and the label.
+
+    The AnnotationTask is supposed to deal with sets and hashable objects, but
+    there are errors transforming the data to sets and using the MASI distance function.
+    For the moment what we do with those type of questions is create a string
+    with all the values, and use the edit distance function.
+
+    Note:
+        We could potentially extend the functionality to a more than a question name. The
+        requirement would be that all the questions are of the same type, as that would
+        determine the type of distance function to use.
+
+    Args:
+        dataset: FeedbackDataset to compute the metrics.
+        question_name: Name of the question for which we want to analyse the agreement.
+
+    Returns:
+        formatted_responses: The responses formatted as a list of tuples of (user_id, question_id, value).
+    """
+    question_type = type(dataset.question_by_name(question_name))
+    supported_question_types = list(QUESTION_TO_DISTANCE.keys())
+    if question_type not in supported_question_types:
+        raise NotImplementedError(
+            f"Question '{question_name}' is of type '{question_type}', the supported question types are: {supported_question_types}."
+        )
+
+    dataset = dataset.filter_by(response_status=ResponseStatusFilter.submitted.value)
+
+    hf_dataset = dataset.format_as("datasets")
```

Comment:
shouldn't we pull first? or does this include filters, sort_by and max_records arguments?

Reply:
Okay, I misunderstood. It is fine then.

Reply:
@plaguss, I realized while sleeping that we do want to include `filter_by`, `sort_by` `max_record` because people need to have control over these feature, for example filtering on the last 1000 record to get the accuracy